frontend/src/Components/MediaUploader.tsx

```diff
@@ -169,86 +247,153 @@ export default function MediaUploader(props: MediaUploaderProps) {
+      } as MediaResult
+    }
+
+    if (uploadData.type === 'video' || uploadData.type === 'image') {
+      return await uploadFile(uploadData.file, uploadData.type)
+    }
+  }
 
+  const uploadFile = async (file: File, type: string) => {
+    try {
       setUploading(true)
 
-      fetch('/upload', {
+      const formData = new FormData()
+      formData.append('file', file)
+
+      const response = await fetch('/upload', {
         method: 'POST',
         body: formData,
       })
-        .then((response) => {
-          setUploading(false)
-
-          console.log('UPLOAD RESPONSE', response)
-
-          if (response.ok) {
-            response
-              .json()
-              .then((data) => {
-                if (data.status === 'ok') {
-                  console.log('UPLOAD COMPLETE', data)
-                  props.onSuccess(process.env.REACT_APP_MEDIA_HOSTING_URL + '/' + data.url, uploadData.type)
-                } else {
-                  console.log('UPLOAD FAILED: no link', data, file.type)
-                  handleError('Произошла ошибка при загрузке 🥺')
-                }
-              })
-              .catch((error) => {
-                console.error('UPLOAD FAILED', error, file.type)
-                handleError('Произошла ошибка при загрузке 🥺')
-              })
-          } else {
-            response
-              .text()
-              .then((text) => {
-                console.error('UPLOAD FAILED', response.status, file.type, text)
-              })
-              .catch((err) => {
-                console.error('UPLOAD FAILED', response.status, file.type, response, err)
-              })
-
-            handleError('Произошла ошибка при загрузке 🥺')
-          }
-        })
-        .catch((error) => {
-          setUploading(false)
-          console.error('UPLOAD FAILED', file.type, error)
-          handleError('Произошла ошибка при загрузке 🥺')
-        })
+
+      console.debug('UPLOAD RESPONSE', response)
+
+      if (!response.ok) {
+        const errorText = await response.text().catch(() => 'Неизвестная ошибка')
+        console.error('UPLOAD FAILED', response.status, file.type, errorText)
+        throw new Error('UPLOAD FAILED: ' + errorText)
+      }
+
+      const data = await response.json()
+
+      if (data.status === 'ok') {
+        console.debug('UPLOAD COMPLETE', data)
+        return {
+          type,
+          url: process.env.REACT_APP_MEDIA_HOSTING_URL + '/' + data.url,
+        } as MediaResult
+      } else {
+        console.error('UPLOAD FAILED: no link', data, file.type)
+        throw new Error('UPLOAD FAILED: no link')
+      }
+    } finally {
+      setUploading(false)
+    }
+  }
+
+  useEffect(() => {
+    if (props.mediaData) {
+      readFile(props.mediaData).then((data) => setUploadArray([...uploadArray, data]))
+    }
+
+    document.addEventListener('paste', handlePaste)
+    return () => {
+      document.removeEventListener('paste', handlePaste)
     }
+  }, [])
```

Comment:
probably need to add `handlePaste` as a dependency here.

also note, that without specifying dependencies the captured `uploadArray` would be always the same here.